messages/3.0.0.md

```diff
@@ -0,0 +1,206 @@
+# MarkdownTOC - 3.0.0
+
+---
+
+**NOTE**
+This upgrade contains some **breaking changes**. 
+Please follow "Upgrade guide from v2 to v3" instructions.
+
+---
+
+## Changes
+
+1. `MarkdownTOC.sublime-settings` format has been changed.
+2. `default_bullets` is deprecated, use `defaults.bullets` instead.
+3. `default_depth` is deprecated, use `levels` instead. Ref: #55
+4. `lowercase_only_ascii` is deprecated, use `lowercase` instead with new value format. Ref: #105
+5. `defaults.bracket`(ex `default_bracket`) default value is changed to `"round"`
+6. Allow regular expressions in 'id_replacements', the format has been changed. Ref: #113
+
+
+# Upgrade Guide
+
+## Upgrading from MarkdownTOC v2.x to v3
+
+First of all `default_xxx` moves to `defaults.xxx`.
+Please modify your settings if you have your own **User Settings**(`Packages/User/MarkdownTOC.sublime-settings`).
+
+
+### 1. `MarkdownTOC.sublime-settings` format has been changed
+
+Old format:
+
+```json
+
+{
+  "default_autoanchor": false,
+  "default_autolink": false,
+  "default_indent": "\t",
+  "default_remove_image": true,
+  "default_link_prefix": "",
+  "default_style": "unordered",
+  "default_uri_encoding": true,
+  "default_markdown_preview": false
+}
+```
+
+New format:
+
+```json
+{
+  "defaults": {
+    "autoanchor": false,
+    "autolink": false,
+    "indent": "\t",
+    "remove_image": true,
+    "link_prefix": "",
+    "style": "unordered",
+    "uri_encoding": true,
+    "markdown_preview": false
+  }
+}
+```
+
+
+### 2. `default_bullets` is deprecated, use `defaults.bullets` instead
+
+Old format:
+
+```json
+{
+  "default_bullets": "-+*"
+}
+```
+
+or value could also be array.
+
+```json
+{
+  "default_bullets": ["-","+","*"]
+}
+```
+
+New format:
+
+```json
+{
+  "defaults": {
+    "bullets": ["-","+","*"]
+  }
+}
+```
+
+Note: Values type is 'array'
+
+You can also override it as attribute
+
+```markdown
+<!-- MarkdownTOC bullets="-,+,*" -->
+```
+
+Note: Values type is 'conmma separated string'
+
+
+### 3. `default_depth` is deprecated, use `defaults.levels` instead
+
+Old format:
+
+```json
+{
+  "default_depth": 0
+}
+```
+
+New format:
+
+```json
+{
+  "defaults": {
+    "levels": [1,2,3,4,5,6]
+  }
+}
+```
+
+Note: Values type is 'array'
+
+You can also override it as attribute
+
+```markdown
+<!-- MarkdownTOC levels="1,2,3" -->
+```
+
+Note: 
+
+- Values type is 'conmma separated string'
```

Comment:
Minor spelling errors: "conmma" should read: "comma" and "string" should be plural as: "strings"

Reply:
OK! thanks 👀